### PR TITLE
hatari: update livecheck and add license

### PR DESCRIPTION
**Pull request by chenrui333**
hatari: update livecheck

### CODE REVIEW

**Comment by chenrui333** [comment]
from `hatari.spec`
```
License:      GPLv2+
```

File: Formula/hatari.rb [hunk 1]
```diff
@@ -5,6 +5,12 @@ class Hatari < Formula
   sha256 "bf48161903930f6470eca1d9c293f51b09b8c54695f5ee213af141316253b8a5"
   head "https://git.tuxfamily.org/hatari/hatari.git"
 
+  livecheck do
+    url "https://hatari.tuxfamily.org/news.html"
+    strategy :page_match
+    regex(/Version v?(\d+(?:\.\d+)+)/i)
```
Comment (samford) [inline Formula/hatari.rb:12]:
```suggestion
    url "https://download.tuxfamily.org/hatari/"
    regex(%r{href=["']?v?(\d+(?:\.\d+)+)/?["' >]}i)
```

**Review by samford — CHANGES_REQUESTED** [review]
The [first-party download page](https://hatari.tuxfamily.org/download.html) simply points to the [directory listing page]((https://download.tuxfamily.org/hatari/)) where the version directories are found, rather than linking to the latest archive file. With that in mind, I think it's best to simply check the directory listing page, as that will align the check with the stable source.

For what it's worth, it's only necessary to use `strategy :page_match` when the URL matches another strategy but we want to specifically use `PageMatch`. These URLs already use `PageMatch`, so we don't explicitly need it here.

File: Formula/hatari.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/hatari.rb]: 👍

**Review by samford — APPROVED** [review]

**Pull request merged**